api/getjobs.go

```diff
@@ -20,7 +22,42 @@ func (api *ImportAPI) getJobsHandler(w http.ResponseWriter, r *http.Request) {
 		logData["filterQuery"] = filtersQuery
 	}
 
-	b, err := api.getJobs(ctx, filterList, logData)
+	offsetParameter := r.URL.Query().Get("offset")
+	limitParameter := r.URL.Query().Get("limit")
+
+	limit := api.defaultLimit
+	offset := api.defaultOffset
+
+	var err error
+
+	if offsetParameter != "" {
+		logData["offset"] = offsetParameter
+		offset, err = utils.ValidatePositiveInt(offsetParameter)
+		if err != nil {
+			log.Event(ctx, "invalid query parameter: offset", log.ERROR, log.Error(err), logData)
+			handleErr(ctx, w, err, nil)
+			return
+		}
+	}
+
+	if limitParameter != "" {
+		logData["limit"] = limitParameter
+		limit, err = utils.ValidatePositiveInt(limitParameter)
+		if err != nil {
+			log.Event(ctx, "invalid query parameter: limit", log.ERROR, log.Error(err), logData)
+			handleErr(ctx, w, err, nil)
+			return
+		}
+	}
+
+	if limit > api.maxLimit {
+		logData["max_limit"] = api.maxLimit
+		log.Event(ctx, "limit is greater than the maximum allowed", log.ERROR, logData)
+		handleErr(ctx, w, errs.ErrInvalidQueryParameter, nil)
```

Comment:
Just a thought but maybe this error could be more informative. There is this example in the permissions-api: https://github.com/ONSdigital/dp-permissions-api/blob/develop/apierrors/errors.go#L17